src/sandbox/enviroments/twenty_forty_eight/env.py

```diff
@@ -0,0 +1,66 @@
+from typing import Optional, Union, Tuple
+from numpy.typing import NDArray
+import gym
+import numpy as np
+import random
+
+from sandbox.enviroments.twenty_forty_eight.game.game import TwentyFortyEightGame
+from sandbox.enviroments.twenty_forty_eight.game.state import TwentyFortyEightState
+from sandbox.enviroments.twenty_forty_eight.game.heuristic import TwentyFortyEightHeuristic
+from sandbox.enviroments.twenty_forty_eight.game.action import TwentyFortyEightPlayerAction, Direction
+
+
+class TwentyFortyEightEnv(gym.Env):
+    def __init__(self, invalid_move_threshold=16, invalid_move_percentage=0.1):
+        self.game = TwentyFortyEightGame()
+        self.heuristic = TwentyFortyEightHeuristic(self.game)
+        self.state = self.game.initial_game_state()
+        self.observation_space = gym.spaces.Box(0, 2048, (self.game.board_dim*self.game.board_dim, ), dtype=np.uint32)
+        self.action_space = gym.spaces.Discrete(4)
+        self.invalid_move_threshold = invalid_move_threshold
+        self.invalid_move_percentage = invalid_move_percentage
+        self._total_count = 0
+        self._invalid_count = 0
+
+    def step(self, action: int) -> Tuple[NDArray, float, bool, dict]:
+        new_state = self._update_state(action)
+        reward = self.heuristic(new_state) - self.heuristic(self.state)
+        done = self.game.is_terminal_state(new_state)
+        if self._too_many_invalid():
+            done = True
+            reward = -np.inf
+        self.state = new_state
+        observation = self.state.board.flatten().astype(np.uint32)
+        info = {}
+        return observation, float(reward), done, info
+
+    def reset(self, *, seed: Optional[int] = None, return_info: bool = False, options: Optional[dict] = None) -> Union[
+        NDArray, tuple[NDArray, dict]]:
+        self.state = self.game.initial_game_state()
+        self._total_count = 0
+        self._invalid_count = 0
+        return self.state.board.flatten().astype(np.uint32)
+
+    def render(self, mode="human"):
+        pass
```

Comment:
I think this method can be supported as game can draw its state but I don't know what are we going to do with visualization.